index.js

```diff
@@ -326,21 +282,23 @@
   //. > odd (42)
   //. false
   //. ```
-  function odd(n) {
-    return n % 2 !== 0;
-  }
+  const odd = n => n % 2 !== 0;
   _.odd = {
     types: [Int, $.Boolean],
-    impl: odd
+    impl: odd,
   };
 
-  var def = $.create ({checkTypes: true, env: $.env});
-  return (Object.keys (_)).reduce (function(int, name) {
-    int[name] = def (name) ({}) (_[name].types) (_[name].impl);
+  {
+    const def = $.create ({checkTypes: true, env: $.env});
+    const int = {Int, NonZeroInt};
+    for (const name of Object.keys (_)) {
+      const {types, impl} = _[name];
+      int[name] = def (name) ({}) (types) (impl);
+    }
     return int;
-  }, {Int: Int, NonZeroInt: NonZeroInt});
+  }
```

Comment:
I prefer this imperative approach, now that we have access to block-scoped variables.

I did not use [`Object.entries`][1] as it was not specified in [ES6][2].


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries
[2]: https://262.ecma-international.org/6.0/